File: DSA Practice/LeetCode/1491. Average Salary Excluding the Minimum and Maximum Salary/solution.java
class Solution {
    public double average(int[] salary) {
        int ma = Integer.MIN_VALUE;
        int mi = Integer.MAX_VALUE;

        int s = 0;
        int c = 0;

        for (int i : salary){

            if (i > ma){
                ma = i;
            }

            if (i < mi){
                mi = i;
            }
        }

        for (int i : salary){
            if (i == ma | i == mi){
                continue;
            }

            s += i;
            c++;
        }

        return (double)s/(double)c;
    }
}

File: DSA Practice/LeetCode/1491. Average Salary Excluding the Minimum and Maximum Salary/solution.py
class Solution:
    def average(self, salary: List[int]) -> float:
        ma = max(salary)
        mi = min(salary)

        s = 0
        c = 0

        for i in salary:
            if i == ma or i == mi:
                continue

            s += i
            c += 1
        
        s /= c

        return s